src/main/java/com/gmail/nossr50/skills/herbalism/HerbalismManager.java

```diff
@@ -757,69 +757,69 @@ private boolean processGreenThumbPlants(BlockState blockState, BlockBreakEvent b
+                seed = Material.CARROT;
                 break;
-
             case "wheat":
-                seed = Material.matchMaterial("WHEAT_SEEDS");
+                seed = Material.WHEAT_SEEDS;
                 break;
-
             case "nether_wart":
-                seed = Material.getMaterial("NETHER_WART");
+                seed = Material.NETHER_WART;
                 break;
-
             case "potatoes":
-                seed = Material.matchMaterial("POTATO");
+                seed = Material.POTATO;
                 break;
-
             case "beetroots":
-                seed = Material.matchMaterial("BEETROOT_SEEDS");
+                seed = Material.BEETROOT_SEEDS;
                 break;
-
             case "cocoa":
-                seed = Material.matchMaterial("COCOA_BEANS");
+                seed = Material.COCOA_BEANS;
                 break;
-
             case "torchflower":
                 seed = Material.matchMaterial("TORCHFLOWER_SEEDS");
                 break;
             default:
                 return false;
         }
 
-
         ItemStack seedStack = new ItemStack(seed);
 
         if (ItemUtils.isAxe(blockBreakEvent.getPlayer().getInventory().getItemInMainHand())
-        && blockState.getType() != Material.COCOA) {
+                && blockState.getType() != Material.COCOA) {
             return false;
         }
 
         if (!greenTerra && !ProbabilityUtil.isSkillRNGSuccessful(SubSkillType.HERBALISM_GREEN_THUMB, player)) {
             return false;
         }
 
-        if (!playerInventory.containsAtLeast(seedStack, 1)) {
+        // Check for seeds in both the main hand and off-hand
+        if (!(playerInventory.containsAtLeast(seedStack, 1) || playerInventory.getItemInOffHand().getType() == seed)) {
+            NotificationManager.sendPlayerInformation(player, NotificationType.REQUIREMENTS_NOT_MET, "Skills.NeedMore", StringUtils.getPrettyItemString(seed));
             return false;
         }
 
+        // Remove seed from the appropriate inventory slot (main hand or off-hand)
+        if (playerInventory.getItemInMainHand().getType() == seed) {
+            playerInventory.removeItem(new ItemStack(seed, 1));
+        } else if (playerInventory.getItemInOffHand().getType() == seed) {
+            playerInventory.getItemInOffHand().setAmount(playerInventory.getItemInOffHand().getAmount() - 1);
+        }
+
+        player.updateInventory(); // Needed until replacement available
+
         if (!processGrowingPlants(blockState, ageable, blockBreakEvent, greenTerra)) {
             return false;
         }
 
-        if(EventUtils.callSubSkillBlockEvent(player, SubSkillType.HERBALISM_GREEN_THUMB, blockState.getBlock()).isCancelled()) {
+        if (EventUtils.callSubSkillBlockEvent(player, SubSkillType.HERBALISM_GREEN_THUMB, blockState.getBlock()).isCancelled()) {
             return false;
         } else {
-            playerInventory.removeItem(seedStack);
```

Comment:
This is where you would remove the seeds, as this `else` block only happens if the outgoing SubSkillEvent is not cancelled